omod/src/main/java/org/openmrs/module/datafilter/extension/html/Content.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.module.datafilter.extension.html;
+
+import java.util.List;
+
+public class    Content {
+
+    private String styles;
+    private String scripts;
+    private String title;
+    private String contentStart;
+    private String contentEnd;
+    private String locationsHtml = "";
+
+
+    public Content() {
+        this.styles = "<style>.listItemBoxCustom {width: 460px;" +
+                "padding: 2px;" +
+                "border: 1px solid lightgray;" +
+                "float: left;" +
+                "background-color: #EFEFEF;" +
+                "overflow-x: scroll;" +
+                "height: 200px;}" +
+                "</style>";
+        this.scripts = "<script type='text/javascript'>" +
+                "let selectedLocationsCount = jQuery('#locationStrings input:checked').length; " +
+                "jQuery('#selectedLocations').text(selectedLocationsCount);" +
+                "</script>";
+        this.title = "Location";
+        this.contentStart = styles + scripts +
+                "<td valign='top'>" + title + "</td>" +
+                "<p>Selected Locations:</p>" + "<p id='selectedLocations'></p>" +
+                "<td valign='top'>" +
+                "<div id='locationStrings' class='listItemBoxCustom'>";
+        this.contentEnd = "</div></td>";
+    }
+
+
+    public void addHTMLForLocationNames(List<String> locationNames) {
+        locationsHtml += locationNames.stream().reduce("", (appendedContent, locationName) -> {
+            appendedContent += "<span class='listItem'>";
+            appendedContent += "<input type='checkbox' name='locationStrings' id='locationStrings." + locationName + "'" + "value='" + locationName + "'" + ">";
```

Comment:
Why not use something like a uuid or location id for the value attribute?

Reply:
We are constructing the view here, it would be difficult for users to grant access if uuid / location id is used